node/src/inherents.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019-2020 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This module is responsible for building the inherent data providers that a Moonbeam node needs.
+//!
+//! A service builder can call the `build_inherent_data_providers` function to get the providers
+//! the node needs based on the parameters it passed in.
+//!
+//! This module also includes MOCK inherent data providers for both the timestamp and validataion
+//! data inherents. Thes emock providers provide stub data that does not represent anything "real"
+//! about the external world, but can pass the runtime's checks. This is useful in testing
+//! for example, running the --dev service without a relay chain backbone, or authoring block
+//! extremely quickly in testing scenarios.
+
+use cumulus_primitives::{
+	inherents::{ValidationDataType, VALIDATION_DATA_IDENTIFIER},
+	PersistedValidationData, ValidationData,
+};
+use parity_scale_codec::Encode;
+use sp_core::H160;
+use sp_inherents::{InherentData, InherentDataProviders, InherentIdentifier, ProvideInherentData};
+use sp_timestamp::{InherentError, INHERENT_IDENTIFIER as TIMESTAMP_IDENTIFIER};
+use std::cell::RefCell;
+
+use cumulus_test_relay_sproof_builder::RelayStateSproofBuilder;
+use moonbeam_runtime::MINIMUM_PERIOD;
+
+/// Build the inherent data providers for the node.
+///
+/// Not all nodes will need all inherent data providers:
+/// - The author provider is entirely optional.
```

Comment:
```suggestion
/// - The author provider is only necessary for block producing nodes
```

*nodes that are participating in staking and want on-chain recognition for block authoring*